src/drm/fairplay.js

```diff
@@ -6,40 +6,41 @@ import {DrmScheme} from './drm-scheme';
 const _logger = getLogger('FairPlay');
 
 const FairPlay: IDrmProtocol = class FairPlay {
+  static scheme: string = DrmScheme.FAIRPLAY;
   /**
    * FairPlay is the configure key system.
-   * @param {Array<Object>} drmData - The drm data.
+   * @param {Array<PKDrmDataObject>} drmData - The drm data.
    * @param {PKDrmConfigObject} drmConfig - The drm config.
    * @return {boolean} - Whether FairPlay is the configure key system.
    */
-  static isConfigured(drmData: Array<Object>, drmConfig: PKDrmConfigObject): boolean {
+  static isConfigured(drmData: Array<PKDrmDataObject>, drmConfig: PKDrmConfigObject): boolean {
     return DrmScheme.FAIRPLAY === drmConfig.keySystem && !!drmData.find(drmEntry => drmEntry.scheme === drmConfig.keySystem);
   }
 
   /**
    * FairPlay playback supports in case 2 conditions are met:
    * 1. The environment supports FairPlay playback.
    * 2. The drm data of the source object contains entry with FairPlay scheme.
-   * @param {Array<Object>} drmData - The drm data to check.
+   * @param {Array<PKDrmDataObject>} drmData - The drm data to check.
    * @return {boolean} - Whether FairPlay can be play on the current environment.
    */
-  static canPlayDrm(drmData: Array<Object>): boolean {
-    _logger.debug('Can play DRM scheme of: ' + DrmScheme.FAIRPLAY);
-    return !!drmData.find(drmEntry => drmEntry.scheme === DrmScheme.FAIRPLAY) && !!window.WebKitMediaKeys;
+  static canPlayDrm(drmData: Array<PKDrmDataObject>): boolean {
+    _logger.debug('Can play DRM scheme of: ' + FairPlay.scheme);
+    return !!drmData.find(drmEntry => drmEntry.scheme === FairPlay.scheme) && !!window.WebKitMediaKeys;
   }
 
   /**
    * Sets the FairPlay playback.
    * @param {FairPlayDrmConfigType} config - The config to manipulate.
-   * @param {Array<Object>} drmData - The drm data.
+   * @param {Array<PKDrmDataObject>} drmData - The drm data.
    * @returns {void}
    */
-  static setDrmPlayback(config: FairPlayDrmConfigType, drmData: Array<Object>): void {
+  static setDrmPlayback(config: FairPlayDrmConfigType, drmData: Array<PKDrmDataObject>): void {
     _logger.debug('Sets drm playback');
-    let drmEntry = drmData.find(drmEntry => drmEntry.scheme === DrmScheme.FAIRPLAY);
+    let drmEntry = drmData.find(drmEntry => drmEntry.scheme === FairPlay.scheme);
     if (drmEntry) {
       config.licenseUrl = drmEntry.licenseUrl;
-      config.certificate = drmEntry.certificate;
+      config.certificate = drmEntry.certificate || '';
```

Comment:
why is this being changed? are we sure empty string is valid?

Reply:
this is a type enforcement because the certificate filed on  PKDrmDataObject type is optional 
https://github.com/kaltura/playkit-js/blob/master/flow-typed/types/drm-data.js#L5
so we have to either fix like the above or remove the optional